File: BEE 1094.py
N = int(input())
C = 0
R = 0
S = 0
for i in range(N):
    a, b = input().split()
    if b == "C":
        C += int(a)
    if b == "R":
        R += int(a)
    if b == "S":
        S += int(a)
total = C + R + S
print("Total: {} cobaias".format(total))
print("Total de coelhos: {}".format(C))
print("Total de ratos: {}".format(R))
print("Total de sapos: {}".format(S))
print("Percentual de coelhos: {0:.2f} %".format((C/total)*100))
print("Percentual de ratos: {0:.2f} %".format((R/total)*100))
print("Percentual de sapos: {0:.2f} %".format((S/total)*100))

File: BEE 1094.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
// var lines = input.split('\n');
var sep = [' ', '\n'];
// console.log(separators.join('|'));
var line = input.split(new RegExp(sep.join('|'), 'g'));

var n = parseInt(line.shift());
var c = 0;
var r = 0;
var s = 0;
for (let i = 0; i < n; i++){
    var a = parseInt(line.shift());
    var b = line.shift();
    if (a >= 1 && a <= 15) {
        if (b == "C") {
            c += a;
        }
        if (b == "R") {
            r += a;
        }
        if (b == "S") {
            s += a;
        }
    }
}
var total = c + r + s;
console.log("Total:", total, "cobaias");
console.log("Total de coelhos:", c);
console.log("Total de ratos:", r);
console.log("Total de sapos:", s);

console.log("Percentual de coelhos:", ((c * 100.0) / total).toFixed(2), "%");
console.log("Percentual de ratos:", ((r * 100.0) / total).toFixed(2), "%");
console.log("Percentual de sapos:", ((s * 100.0) / total).toFixed(2), "%");